### DISCUSSION TITLE
Resource Route Issues with Updating and Deleting

### DISCUSSION THREAD

**Discussion by kayGweb** [adonisjs/core#2266] [Help] [answered]
So I really like the concept of Resource routes, it saves a lot of time.  

However, I have issues when attempting to use the update and Delete routes.  I have the form listed below.  I used the command adonis route:list and the resource route seems to have created routes as it should, however still not working.  When I submit the below form It sends a GET request and that activates the AdminVideoController.show endpoint.  Any help is appreciated!

form class="form" role="form" method="PUT" **action="{{ route('AdminVideoController.update', {id: video.id }) }}"** autocomplete="off"

The result of the above edge template is this:
form class="form" role="form" method="PUT" action="/admin/video/5" autocomplete="off"
   

            <div class="form-group row">
                <label class="col-lg-3 col-form-label form-control-label"></label>
                <div class="col-lg-9">
                    {{--  <a href="{{ route('back') }}" type="reset" class="btn btn-secondary" value="Clear"> Back</a>  --}}
                    <input type="submit" formmethod="PATCH" class="btn btn-primary" value="Save Changes">
                </div>
            </div>
        </form>

**Here is the list of routes from the adonis route:list command**

/admin/video                   │ HEAD,GET  │ AdminVideoController.index                  │ /admin/video.index            

/admin/video/create     │ HEAD,GET     │ AdminVideoController.create                 │ /admin/video.create           

/admin/video                 │ POST            │ AdminVideoController.store                  │ /admin/video.store            

 /admin/video/:id           │ HEAD,GET    │ AdminVideoController.show                   │ /admin/video.show             

 /admin/video/:id/edit    │ HEAD,GET    │ AdminVideoController.edit                    │ /admin/video.edit             

 /admin/video/:id           │ PUT,PATCH  │ AdminVideoController.update                │ /admin/video.update           

 /admin/video/:id          │ DELETE         │ AdminVideoController.destroy                │ /admin/video.destroy

**Comment by RomainLanz** [comment, marked as answer]
Hey @kayGweb!

The `<form>` tag only handle `GET` or `POST` request.

What you are looking for is called `Method Spoofing`.
We haven't yet documented this in the V5 documentation (it's available on [V4 website](https://adonisjs.com/docs/4.1/request#_method_spoofing)).

If you are using AdonisJS 5 and you want to activate method spoofing, you need to modify the settings inside the `config/app.ts` file.

```ts
// ...

export const http: HttpConfig = {
  /*
  |--------------------------------------------------------------------------
  | Allow method spoofing
  |--------------------------------------------------------------------------
  |
  | Method spoofing enables defining custom HTTP methods using a query string
  | `_method`. This is usually required when you are making traditional
  | form requests and wants to use HTTP verbs like `PUT`, `DELETE` and
  | so on.
  |
  */
  allowMethodSpoofing: false,

  // ...
```

**Comment by kayGweb** [comment]
Hey @RomainLanz 

Thanks for your response!  I am using Adonis 4.1 and this has helped tremendously!  My Route.js file is much smaller now!
